### PR TITLE
Update Kubernetes to v1.23

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

* ~Move containerd out to a separate binary~
  Moved to https://github.com/k3s-io/k3s/pull/4757
* ~Fix CI issues~
    Moved out to https://github.com/k3s-io/k3s/pull/4801
* Update Kubernetes to 1.23
* Update golang to 1.17

#### Types of Changes ####

Upstream version change

#### Verification ####

`kubectl --version`

#### Linked Issues ####

* https://github.com/k3s-io/k3s/issues/4689

#### User-Facing Change ####
```release-note
Move containerd out to separate binary
Update Kubernetes to v1.23.1
```

### CODE REVIEW

**Comment by dweomer** [comment]
Would like to see the split out of containerd as a separate binary in its own PR (that gets merged before this kind of bump).

File: pkg/static/zz_generated_bindata.go
Comment (dereknola) [inline pkg/static/zz_generated_bindata.go:5]: Did go change their tags with the new version?

**Review by dereknola — APPROVED** [review]
LGTM

File: pkg/static/zz_generated_bindata.go
Comment (brandond) [reply pkg/static/zz_generated_bindata.go:5]: yes, go 1.17 now automatically adds these build tags for you. I'm not a big fan but I assume they have their reasons.

**Comment by olljanat** [comment]
I understand that separate binary is needed because opentelemetry but is containerd 1.6.0 release really so far that K3s 1.23 versions cannot wait that?

I was hoping that we would be able to solve https://github.com/k3s-io/k3s/issues/4545 on 1.23 and I assume that other RKE2 users would like to see it too.

**Comment by brandond** [comment]
containerd v1.6.0 isn't out yet; I have no idea what their target date is for releasing that. Even if it did come out tomorrow, we would probably be somewhat reluctant to ship a .0 release.

File: scripts/build
Comment (dweomer) [inline scripts/build:15]: The names/values for these two seem transposed.

**Review by dweomer — CHANGES_REQUESTED** [review]
- The **k3s-root** bump should be in a separate PR. _I want to see it as a standalone change in the git log as well as tested end-to-end on its own._
- The **containerd** split should in in a separate PR that includes an ADR. As with k3s-root, _I want to see it as a standalone change in the git log as well as tested end-to-end on its own._

**Comment by brandond** [comment]
> * The **k3s-root** bump should be in a separate PR. _I want to see it as a standalone change in the git log as well as tested end-to-end on its own._
> * The **containerd** split should be in a separate PR that includes an ADR. As with k3s-root, _I want to see it as a standalone change in the git log as well as tested end-to-end on its own._

Sorry, I thought based on our conversation in standup the other day that we were OK not doing these separately since no-one else cared to see them split out.  I can take a moment to create additional PRs for those, but since they are required to successfully build 1.23, I will also leave those commits here and then rebase on top of them once they're merged.

**Review by galal-hussein — APPROVED** [review]

File: scripts/codespell.sh
Comment (Oats87) [inline scripts/codespell.sh:5]: We should definitely make sure to get this into our other branches

**Review by Oats87 — APPROVED** [review]
LGTM; should we torch the vendor directory ....?

File: go.mod
Comment (dweomer) [inline go.mod:16]: gonna bump cri-tools later?

**Review by dweomer — APPROVED** [review]
LGTM (assuming the cri-tools mismatch is on purpose and will be dealt with via follow-on PR)

File: go.mod
Comment (brandond) [reply go.mod:16]: Upstream is having issues, they haven't been able to push a 1.23 tag of cri-tools yet.

File: cmd/k3s/main.go
Comment (brandond) [reply cmd/k3s/main.go]: I think this it on an outdated commit?

Comment (Oats87) [reply cmd/k3s/main.go]: You are correct, I made it earlier when I was battling my browser trying to review this PR and I guess it "saved as draft or something"

**Review by briandowns — APPROVED** [review]

**Pull request merged**
